SIMSBIOHUB-614: Fix file upload error handling.

NickPhura opened the pull request:
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-614?

## Description of Changes

Split the project/survey report/non-report attachment dialogs into 2 components.
Made 2 new attachment dialogs - one for single items, one for many items.
Added error handling to catch file upload errors.
Moved a few files around to group like-components.

## Testing Notes

Changes to Project/Survey Report and Non-Report attachment upload
- Should work as before, and should handle any errors thrown during upload.

Changes to observation/telemetry/sample-site/animal csv import
- Should work as before, and should handle any errors thrown during upload.

NickPhura committed:
Split the project/survey report/non-report attachment dialogs into 2 components.
Made 2 new attachment dialogs - one for single items, one for many items.
Added error handling to catch file upload errors.

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-614

mauberti-bc committed:
Merge branch 'dev' into SIMSBIOHUB-614

mauberti-bc commented on app/src/components/attachments/FileUploadWithMeta.tsx:
```diff
@@ -1,84 +1,52 @@
-export const FileUploadWithMeta: React.FC<IFileUploadWithMetaProps> = (props) => {
-  const { handleSubmit, setFieldValue, errors } = useFormikContext<IReportMetaForm>();
+  const onStatus = (status: UploadFileStatus) => {
+    console.log(status);
```
Deliberate console.log?

NickPhura replied:
Nope! will remove

mauberti-bc reviewed (dismissed):
Tested locally and works well 👍 

Minor future UI improvement: Closing the error dialog should also close the upload attachment dialog to avoid displaying "Complete" or "Success" on the upload dialog despite an error.

![image](https://github.com/user-attachments/assets/c578be33-b610-4d3b-b994-be9d36048433)

NickPhura committed:
Add missing JSDoc.
Fix project/survey regular attachments error display (shows error in the list, not as a popup).

NickPhura committed:
Add JSDoc.
Add fallback catch to report upload.
Fix error popup not displaying error message in body of dialog for the csv upload dialogs.

NickPhura commented:
@mauberti-bc 
> Minor future UI improvement: Closing the error dialog should also close the upload attachment dialog to avoid displaying "Complete" or "Success" on the upload dialog despite an error.

The error popup was actually a copy/paste mistake. The file upload component handles individual file errors already, so I just had to remove the try/catch and let it bubble up:

![image](https://github.com/user-attachments/assets/a091ebf8-0a4e-40d6-8938-4cd4b8def25b)

Tweaked some other trivial stuff as well, and added some JSDoc. So should all be happy now.

NickPhura committed:
Merge branch 'dev' into SIMSBIOHUB-614

mauberti-bc reviewed (approved):
👍
